trade-finance-manager-ui/server/controllers/case/parties/partyUrnValidation.validate.test.js

```diff
@@ -0,0 +1,91 @@
+    expect(response.errorsObject.errors.fieldErrors).toEqual({ partyUrn: { text: 'Enter a correct party URN' } });
+    expect(response.urn).toEqual('1');
+  });
+
+  it('should return validation error if partyUrn is a letter', () => {
+    const urn = 'AA';
+
+    const response = validatePartyURN(urn);
+    expect(response.errorsObject.errors.errorSummary).toEqual([{ text: 'Enter a correct party URN', href: '#partyUrn' }]);
+    expect(response.errorsObject.errors.fieldErrors).toEqual({ partyUrn: { text: 'Enter a correct party URN' } });
+    expect(response.urn).toEqual('AA');
+  });
+
+  it('should return validation error if partyUrn is 2 numbers', () => {
+    const urn = '11';
+
+    const response = validatePartyURN(urn);
+    expect(response.errorsObject.errors.errorSummary).toEqual([{ text: 'Enter a correct party URN', href: '#partyUrn' }]);
+    expect(response.errorsObject.errors.fieldErrors).toEqual({ partyUrn: { text: 'Enter a correct party URN' } });
+    expect(response.urn).toEqual('11');
+  });
+
+  it('should return validation error if partyUrn is letters and numbers', () => {
+    const urn = 'ABC123';
+
+    const response = validatePartyURN(urn);
+    expect(response.errorsObject.errors.errorSummary).toEqual([{ text: 'Enter a correct party URN', href: '#partyUrn' }]);
+    expect(response.errorsObject.errors.fieldErrors).toEqual({ partyUrn: { text: 'Enter a correct party URN' } });
+    expect(response.urn).toEqual('ABC123');
+  });
+
+  it('should return validation error if partyUrn has special characters', () => {
+    const urn = 'A"1';
+
+    const response = validatePartyURN(urn);
+    expect(response.errorsObject.errors.errorSummary).toEqual([{ text: 'Enter a correct party URN', href: '#partyUrn' }]);
+    expect(response.errorsObject.errors.fieldErrors).toEqual({ partyUrn: { text: 'Enter a correct party URN' } });
+    expect(response.urn).toEqual('A"1');
+  });
+
+  it('should return validation error if partyUrn is only special characters', () => {
+    const urn = '"!£!"£!"£!"£';
+
+    const response = validatePartyURN(urn);
+    expect(response.errorsObject.errors.errorSummary).toEqual([{ text: 'Enter a correct party URN', href: '#partyUrn' }]);
+    expect(response.errorsObject.errors.fieldErrors).toEqual({ partyUrn: { text: 'Enter a correct party URN' } });
+    expect(response.urn).toEqual('"!£!"£!"£!"£');
+  });
+
+  it('should return validation error if partyUrn is letters and numbers', () => {
+    const urn = '12C';
+
+    const response = validatePartyURN(urn);
+    expect(response.errorsObject.errors.errorSummary).toEqual([{ text: 'Enter a correct party URN', href: '#partyUrn' }]);
+    expect(response.errorsObject.errors.fieldErrors).toEqual({ partyUrn: { text: 'Enter a correct party URN' } });
+    expect(response.urn).toEqual('12C');
+  });
+
+  it('should return validation error if partyUrn is correct length of numbers but has a special character', () => {
```

Comment:
Can we please add a case with an empty space input please.

Reply:
done